cmd/service_bootstrap_desktop.go

```diff
@@ -134,7 +134,7 @@ func (di *Dependencies) bootstrapServiceOpenvpn(nodeOptions node.Options) {
 		proposal := openvpn_discovery.NewServiceProposalWithLocation(currentLocation, transportOptions.Protocol)
 
 		var portSupplier port.ServicePortSupplier = di.PortPool
-		if transportOptions.Port != 0 {
+		if transportOptions.Port != 0 && locationInfo.PubIP == locationInfo.OutIP {
```

Comment:
@zolia is not sure why we removed this condition. Because it port suplly

Reply:
@Waldz it was a bug that @zolia himself fixed in https://github.com/mysteriumnetwork/node/pull/896/commits/82f0278834a0cbe0b130f01c7e5e63db81093da0 
With this condition, `--openvpn.port` is not respected for users under NAT (because, obviously, their IP does not match external IP).

Reply:
Would be good to honor Provider's which from CLI flag.
ELSE
Defaults to random port

Reply:
So this:
```golang
if transportOptions.Port != 0 && locationInfo.PubIP == locationInfo.OutIP {
    portSupplier = port.NewFixed(transportOptions.Port)
}
```

becomes:

```golang
if transportOptions.Port != 0  {
    portSupplier = port.NewFixed(transportOptions.Port)
} else if locationInfo.PubIP == locationInfo.OutIP {
    portSupplier = // default or random, you decide.
}
```

where's the issue?